module-2/backend/week_three.md

```diff
@@ -9,17 +9,80 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 
 ### Questions
 
-1. What are the 3 main features in an ERD?
+1. What are the 3 main features in an ERD? ERD is a good tool to visualize the schema of the database.
+
+It shows what columns/attributes there are in tables/models. More importantly, it shows primary keys and foreign keys and therefore is a good indicator of one-to-one and/or one-to-many relationships. Finally, it demonstrates joint tables which stand for many-to-many relationsihps. 
+
 2. Create an ERD for the following database: Restaurants, Customers, Items, Ingredients, Beverages, Orders, Vendors.
+
+    restaurants: id, name, address, phone_number
+    customers:   id, name, phone_number 
+      
+    restaurant-customers: id, restaurant_id, customer_id
+    
+    ingredients: id, title, item_id
+    items:       id, title, order_id
+    beverages:   id, title, type, order_id
+    orders:      id, customer_id
+    
+    vendors:     id, name, phone_number, email
+    
+    restaurant-vendors: id, restaurant_id, vendor_id
+    
+  Sorry, didn't know how to draw in markdown. In other words, a customer can go to many restaurants, and a restaurant has many             customers. An ingredient belongs to an item, an item and a beverage both belong to an order. A vendor has many restaurants it delivers   goods/services to, and a restaurant has many vendors.
+    
+
 3. What is the entry at the command line to create a new rails app?
+  
+  rail new myapp
+
 4. What do Models generally inherit from in rails?
+  
+  ActiveRecord::Base
+
 5. What do Controllers generally inherit from in a rails project?
+
+  ApplicationController
+  
 6. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+
+  get '/horses/:id', to: 'horses#show' 
+  Alternatively: resources: horses, only: [:show]
+  
 7. What rake task is useful when looking at routes, and what information does it give you?
+
+  rake routes shows all routes available within the app
```

Comment:
True, but a little more specifically, it gives you your prefixes, verbs, URI patterns, and controller actions.